programs/keeper-client/KeeperClient.cpp

```diff
@@ -0,0 +1,333 @@
+    else
+        history_file = home_path + "/.keeper-client-history";
+
+    if (!history_file.empty() && !fs::exists(history_file))
+    {
+        try
+        {
+            FS::createFile(history_file);
+        }
+        catch (const ErrnoException & e)
+        {
+            if (e.getErrno() != EEXIST)
+                throw;
+        }
+    }
+
+    Poco::Logger::root().setLevel(config().getString("log-level", "error"));
+
+    EventNotifier::init();
+}
+
+void KeeperClient::executeQuery(const String & query)
+{
+    std::vector<String> queries;
+    boost::algorithm::split(queries, query, boost::is_any_of(";"));
+
+    for (const auto & query_text : queries)
+    {
+        if (!query_text.empty())
+            processQueryText(query_text);
+    }
+}
+
+bool KeeperClient::processQueryText(const String & text)
+{
+    if (exit_strings.find(text) != exit_strings.end())
+        return false;
+
+    std::vector<String> tokens;
+    boost::algorithm::split(tokens, text, boost::is_any_of(" "));
+
+    try
+    {
+        if (need_confirmation)
+        {
+            need_confirmation = false;
+            if (tokens.size() == 1 && (tokens[0] == "y" || tokens[0] == "Y"))
+                confirmation_callback();
+        }
+        else if (tokens.size() == 1 && tokens[0].size() == 4 && four_letter_word_commands.find(tokens[0]) != four_letter_word_commands.end())
+            std::cout << executeFourLetterCommand(tokens[0]) << "\n";
+        else
+        {
+            auto callback = commands.find({tokens[0], tokens.size() - 1});
+            if (callback == commands.end())
+            {
+                if (tokens[0].size() == 4 && tokens.size() == 1)  /// Treat it like unrecognized four-letter command
+                    std::cout << executeFourLetterCommand(tokens[0]) << "\n";
+                else
+                    std::cerr << "No command found with name " << tokens[0] << " and args count " << tokens.size() - 1 << "\n";
+            }
+            else
+                callback->second(this, tokens);
+        }
+    }
+    catch (Coordination::Exception & err)
+    {
+        std::cerr << err.message() << "\n";
+    }
+    return true;
+}
+
+void KeeperClient::runInteractive()
+{
+
+    LineReader::Patterns query_extenders = {"\\"};
+    LineReader::Patterns query_delimiters = {};
+
+    ReplxxLineReader lr(suggest, history_file, false, query_extenders, query_delimiters, {});
+    lr.enableBracketedPaste();
+
+    while (true)
+    {
+        String prompt;
+        if (need_confirmation)
+            prompt = "[y/n] ";
+        else
+            prompt = cwd.string() + " :) ";
+
+        auto input = lr.readLine(prompt, ":-] ");
+        if (input.empty())
+            break;
+
+        if (!processQueryText(input))
+            break;
+    }
+}
+
+int KeeperClient::main(const std::vector<String> & args)
+{
+    if (args.empty())
+        zk_args.hosts = {"localhost:2181"};
```

Comment:
@pufit It would be nice to have a separate options `--host` and `--port` to be consistent with `clickhouse-client`

Reply:
Oops, I forgot to remove this!